Add Ethernet.setHostname method and make DHCP hostname configurable.

dlahoza opened the pull request:
- Added Ethernet.setHostname(const char* hostname) method.
- Increased DHCP buffer size to 64 bytes to accommodate variable hostname length.
- Updated DhcpClass to use the custom hostname.

dlahoza committed:
Add Ethernet.setHostname method and make DHCP hostname configurable.

- Added Ethernet.setHostname(const char* hostname) method.
- Increased DHCP buffer size to 64 bytes to accommodate variable hostname length.
- Updated DhcpClass to use the custom hostname.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino-libraries/Ethernet?pullRequest=297) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino-libraries/Ethernet?pullRequest=297) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino-libraries/Ethernet?pullRequest=297) it.</sub>

dlahoza committed:
Add DhcpCustomHostname example

This example demonstrates how to use the `Ethernet.setHostname()` function to set a custom hostname for DHCP requests. It is based on the `DhcpAddressPrinter` example.
